lib/syskit/exceptions.rb

```diff
@@ -483,6 +483,39 @@ def pretty_print(pp)
         end
     end
 
+    class ConflictingDeploymentAllocation < SpecError
+        attr_reader :deployment_to_tasks
+
+        def initialize(deployment_to_tasks)
+            @deployment_to_tasks = deployment_to_tasks
+        end
+
+        def pretty_print(pp)
+            pp.text "cannot deploy the following tasks"
+            deployment_to_tasks.each do |deployed_task, tasks|
+                tasks.each do |task|
+                    pp.nest(2) do
+                        pp.breakable
+                        pp.text "#{task} (#{task.orogen_model.name})"
+                    end
+                end
+                pp.breakable
+                pp.text "because the same "
+                process_server_name = deployed_task.configured_deployment
+                                                   .process_server_name
+                orogen_model = deployed_task.configured_deployment
+                                            .orogen_model
+                pp.text(
+                    "deployed task #{deployed_task.mapped_task_name} from deployment " \
+                    "#{orogen_model.name} is defined in " \
+                    "#{orogen_model.project.name} on #{process_server_name}"
```

Comment:
That a task is defined somewhere else does not impede anything.

The issue is that the same deployment is used multiple times. Since we demand that all deployments are given in the deployment stage, you should deploy the task and the corresponding toplevel tasks (i.e. explain why these tasks are in the plan)

I recommend discussing this with @jhonasiv, he has a very good understanding now of the relationship between tasks and instance requirements :P